tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/do_not_track.py

```diff
@@ -9,27 +9,33 @@
 
 class DoNotTrack(Base):
 
-    _allow_tracking_checkbox_locator = (By.XPATH, '//li/label[span[@data-l10n-id="allowTracking"]]')
-    _disallow_tracking_checkbox_locator = (By.XPATH, '//li/label[span[@data-l10n-id="doNotTrackActions"]]')
-    _do_not_have_pref_on_tracking_checkbox_locator = (By.XPATH, '//li/label[span[@data-l10n-id="doNotHavePrefOnTracking"]]')
+    _allow_tracking_radio_locator = (By.CSS_SELECTOR, 'gaia-radio[name="privacy.donottrackheader.value"][value="0"]')
+    _disallow_tracking_radio_locator = (By.CSS_SELECTOR, 'gaia-radio[name="privacy.donottrackheader.value"][value="1"]')
+    _do_not_have_pref_on_tracking_radio_locator = (By.CSS_SELECTOR, 'gaia-radio[name="privacy.donottrackheader.value"][value="-1"]')
 
     def tap_allow_tracking(self):
-        label = self.marionette.find_element(
-            *self._allow_tracking_checkbox_locator)
-        label.tap()
-        radio = label.find_element(By.TAG_NAME, 'input')
-        Wait(self.marionette).until(expected.element_selected(radio))
+        element = self.marionette.find_element(
+            *self._allow_tracking_radio_locator)
+        element.tap()
+        # The following should work, but doesn't, see bug 1113742. We use execute_script instead, for now
+        # Wait(self.marionette).until(expected.element_selected(element))
+        Wait(self.marionette).until(lambda m:
```

Comment:
We use the workaround also in Keyboard and Settings. I'd suggest to factorize it in `Base.py`, like:
```python
 # TODO: Remove me once bug 1113742 is fixed
def wait_on_check_state_for_shadow_dom_element(self, element, checked=True):
    Wait(self.marionette).until(lambda m:
            m.execute_script("return arguments[0].wrappedJSObject.checked", [element]) is checked)
```

and then here:
```python
element.tap()
self.wait_on_check_state_for_shadow_dom_element(element)
```